python-package/xgboost/compat.py

```diff
@@ -55,20 +56,43 @@ def lazy_isinstance(instance: Any, module: str, name: str) -> bool:
         from sklearn.cross_validation import KFold as XGBKFold
         from sklearn.cross_validation import StratifiedKFold as XGBStratifiedKFold
 
+    # sklearn.utils Tags types can be imported unconditionally once
```

Comment:
I don't think we should.

That'd effectively raise `xgboost`'s requirement all the way to `scikit-learn>=1.6`.

Because it would result in `compat.SKLEARN_INSTALLED` being `False` for `scikit-learn < 1.6`:

https://github.com/dmlc/xgboost/blob/5826b02d036deae8c232fb35654eb21f57490919/python-package/xgboost/compat.py#L60-L61

Which would make all the estimators unusable on those versions.

https://github.com/dmlc/xgboost/blob/5826b02d036deae8c232fb35654eb21f57490919/python-package/xgboost/sklearn.py#L754-L757